src/backend/distributed/metadata/node_metadata.c

```diff
@@ -757,6 +772,20 @@ master_update_node(PG_FUNCTION_ARGS)
 		if (force)
 		{
 			handle = StartLockAcquireHelperBackgroundWorker(MyProcPid, lock_cooldown);
+			if (!handle)
+			{
+				/*
+				 * We failed to start a background worker, which probably means that we exceeded
+				 * max_worker_processes, and this is unlikely to be resolved by retrying. We do not want
+				 * to repeatedly throw an error because if master_update_node is called to complete a
+				 * failover then finishing is the only way to bring the cluster back up. Therefore we
+				 * give up on killing other backends and simply wait for the lock. We do set
+				 * lock_timeout to lock_cooldown, because we don't want to wait forever to get a lock.
+				 */
+				SetLockTimeoutLocally(lock_cooldown);
+				ereport(NOTICE, (errmsg("No background worker slot available."
```

Comment:
could be more explicit in NOTICE message I think. I thought something like below, but you might think of a better alternative:
https://github.com/citusdata/citus/pull/4214#discussion_r499562241

what do you think ?

Reply:
>ereport(WARNING, (errmsg("could not start lock acquiring background worker to "
                         "force the update. Degrading to acquiring locks "
                         "with a lock time out."),
                  errhint("Increasing max_worker_processes might help.")));

Actually to me `Lock acquiring background worker` is not clear.

What about:
```c
ereport(WARNING, (errmsg("could not start background worker to kill backends with conflicting locks to"
                         "force the update. Degrading to acquiring locks "
                         "with a lock time out."),
                  errhint("Increasing max_worker_processes might help."))); 
```

I think it would make sense to refactor this `lock acquiring background worker` to `conflicting lock killler background worker` or something to be more clear. But definitely that should be done in a separate PR

Reply:
this seems better to me

Reply:
added. thank you both!